src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -227,10 +230,14 @@ contract StrategyManagerUnitTests is Test {
         cheats.stopPrank();
     }
 
-    function testDepositIntoStrategySuccessfully(address staker, uint256 amount) public filterFuzzedAddressInputs(staker) {
-        IStrategy strategy = dummyStrat;
+    function testDepositIntoStrategySuccessfully(IStrategy strategy, address staker, uint256 amount) public filterFuzzedAddressInputs(staker) {
         IERC20 token = dummyToken;
 
+        /// IMPORTANT: ensure that when using this test as a part of another test, that the input strategy is already whitelisted!
+        if (!strategyManager.strategyIsWhitelistedForDeposit(strategy)){
+            strategy = dummyStrat;
+        }
+
```

Comment:
this is some questionable test design right here. we shouldn't be modifying inputs like this -- maybe make an internal version of this test that takes a strategy as input and a public version that hard-codes the strategy as `dummyStrat` instead?
This feels like it is a massive footgun that doesn't need to exist

Reply:
yeah I made a comment on it too.  Didn't want to add more duplicate code to an already bloated file, but its not really great.

Reply:
I feel am not comfortable ever merging code that looks like this. Code bloat is 100% preferable to booby-trapped code.

Reply:
fixed it